src/Aspire.Hosting/ApplicationModel/ResourceNotificationService.cs

```diff
@@ -493,9 +493,7 @@ private static CustomResourceSnapshot GetCurrentSnapshot(IResource resource, Res
             {
                 ResourceType = resource.GetType().Name,
                 Properties = [],
-                HealthStatus = resource.TryGetAnnotationsIncludingAncestorsOfType<HealthCheckAnnotation>(out _)
-                    ? null // Indicates that the resource has health check annotations but the health status is unknown.
-                    : HealthStatus.Healthy
```

Comment:
It seems like an incorrect assumption to deem any resource that does not have a health check healthy?

Reply:
Correct. We needed to differentiate between the, I don't have any health status ever and I will have one eventually (it hasn't come in yet). I think it just means, no health reports means the health status isn't available, and at least one means there is.

Reply:
Is differentiation needed? In both cases (I don't have any health status ever) and (I will have one eventually), the current health status are both null

Reply:
If a resource has no heath checks but is running, isn't it by definition healthy based on the criteria supplied by the hosting integration author?

Maybe no health checks + running = healthy
Not running + no health checks = null state.

Reply:
I think by changing this code we would break WaitUntilHealthyAsync(...)

Reply:
> If a resource has no heath checks but is running, isn't it by definition healthy based on the criteria supplied by the hosting integration author?

That could be an assumption we make, since without healthchecks we don't have any way to tell the state of the resource. If so, we could default to `Healthy` but keep `HealthReportSnapshot.HealthStatus` nullable.

Reply:
I've made this change.